lib/table_of_contents/parser.rb

```diff
@@ -44,8 +44,11 @@ def parse_content
         @doc.css(toc_headings).each do |node|
           text = node.text
           id = text.downcase
-          id.gsub!(PUNCTUATION_REGEXP, '') # remove punctuation
+          id.gsub!(PUNCTUATION_REGEXP, ' ') # replace punctuation with spaces
+          id.gsub!(/\s+/,' ') # in case there were two or more special characters
           id.gsub!(' ', '-') # replace spaces with dash
+          id.gsub!(/-+/, '-') # in case there were two or more dashes
+          id.chomp!('-') # not at the end
```

Comment:
Why did you change this id generation logic??

I tried the below toc generation and it does work. 

## heading

```

## a b  c   d

## foo-bar

## foo bar

## a--b---c
```

result:

```
<ul class="section-nav">
<li class="toc-entry toc-h2"><a href="#a-b--c---d">a b  c   d</a></li>
<li class="toc-entry toc-h2"><a href="#hoge-foo">hoge-foo</a></li>
<li class="toc-entry toc-h2"><a href="#hoge-foo-1">hoge foo</a></li>
<li class="toc-entry toc-h2"><a href="#abc">a–b—c</a></li>
<li class="toc-entry toc-h2"><a href="#a-b-c">a b c</a></li>
</ul>
``` 

What toc case do you want to care?

Reply:
I think it breaks in cases when the title has, for example, a slash (`/`) or a question mark (`?`)

Reply:
I think it works although the title has slash or question mark.

```
## abc

## a/b/c

## abc?
```

```html
<li class="toc-entry toc-h2"><a href="#abc">abc</a></li>
<li class="toc-entry toc-h2"><a href="#abc-1">a/b/c</a></li>
<li class="toc-entry toc-h2"><a href="#abc-2">abc?</a></li>
```

This works correctly.